code/game/machinery/iv_drip.dm

```diff
@@ -45,45 +54,67 @@
 	QDEL_NULL(reagent_container)
 	return ..()
 
+/obj/machinery/iv_drip/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(has_gravity())
+		playsound(src, 'sound/effects/roll.ogg', 100, TRUE)
+
 /obj/machinery/iv_drip/ui_interact(mob/user, datum/tgui/ui)
 	ui = SStgui.try_update_ui(user, src, ui)
 	if(!ui)
-		ui = new(user, src, "IVDrip")
+		ui = new(user, src, "IVDrip", name)
 		ui.open()
 
 /obj/machinery/iv_drip/ui_data(mob/user)
 	var/list/data = list()
 	data["transferRate"] = transfer_rate
-	data["injectOnly"] = inject_only ? TRUE : FALSE
+	data["transferStep"] = IV_TRANSFER_RATE_STEP
 	data["maxInjectRate"] = MAX_IV_TRANSFER_RATE
 	data["minInjectRate"] = MIN_IV_TRANSFER_RATE
 	data["mode"] = mode == IV_INJECTING ? TRUE : FALSE
 	data["connected"] = attached ? TRUE : FALSE
-	data["beakerAttached"] = reagent_container ? TRUE : FALSE
+	if(attached)
+		data["objectName"] = attached.name
+	data["injectOnly"] = inject_only || (attached && !isliving(attached)) ? TRUE : FALSE
+	data["containerAttached"] = reagent_container ? TRUE : FALSE
+	var/datum/reagents/drip_reagents = get_reagents()
+	if(drip_reagents)
+		data["containerCurrentVolume"] = round(drip_reagents.total_volume, IV_TRANSFER_RATE_STEP)
+		data["containerMaxVolume"] = drip_reagents.maximum_volume
+		data["containerReagentColor"] = mix_color_from_reagents(drip_reagents.reagent_list)
 	data["useInternalStorage"] = use_internal_storage
+	data["isContainerRemovable"] = !use_internal_storage && !istype(src, /obj/machinery/iv_drip/saline)
```

Comment:
Wait why is this like this?

Reply:
the saline drip has a non-removable saline tank, that is not an internal storage. Yeah it's a bit weird why they didn't use the internal container, but I decided to not touch that because I don't know the reasoning.

Reply:
The plumbing one is weird too. It injects chemicals avoiding its internal container, directly from the network. And it stores the blood in this container. So you can inject chemicals and draw blood using one plumbing drip. But again, maybe there is a reason for this.

Reply:
I know the reason for the plumbing one, and that's because IV was made to support only one internal beaker and otherwise you couldn't both inject and extract at the same time because it'd end up/use the same beaker.

In the future just add a var/can_remove_container for this, because an arbitrary istype check like this is not great